servlet/security-clientcert/src/test/java/org/javaee7/servlet/security/clientcert/SecureServletWithCommonNameTest.java

```diff
@@ -0,0 +1,286 @@
+        if (baseHttps == null) {
+            throw new IllegalStateException("No https URL could be created from " + base);
+        }
+        
+        // ### Ask the server for its certificate and add that to a new local trust store
+        
+        // Server -> client : the trust store certificates are used to validate the certificate sent
+        // by the server
+        
+        X509Certificate[] serverCertificateChain = getCertificateChainFromServer(baseHttps.getHost(), baseHttps.getPort());
+        
+        if (!isEmpty(serverCertificateChain)) {
+            
+            System.out.println("Obtained certificate from server. Storing it in client trust store");
+        
+            String trustStorePath = createTempJKSTrustStore(serverCertificateChain);
+    
+            System.out.println("Reading trust store from: " + trustStorePath);
+            
+            webClient.getOptions().setSSLTrustStore(new File(trustStorePath).toURI().toURL(), "changeit", "jks");
+            
+            // If the use.cnHost property is we try to extract the host from the server
+            // certificate and use exactly that host for our requests.
+            // This is needed if a server is listening to multiple host names, for instance
+            // localhost and example.com. If the certificate is for example.com, we can't
+            // localhost for the request, as that will not be accepted.
+            if (System.getProperty("use.cnHost") != null) {
+                System.out.println("use.cnHost set. Trying to grab CN from certificate and use as host for requests.");
+                baseHttps = getHostFromCertificate(serverCertificateChain, baseHttps);
+            }
+        } else {
+            System.out.println("Could not obtain certificates from server. Continuing without custom truststore");
+        }
+       
+        System.out.println("Using client key store from: " + clientKeyStorePath);
+
+        // Client -> Server : the key store's private keys and certificates are used to sign
+        // and sent a reply to the server
+        webClient.getOptions().setSSLClientCertificate(new File(clientKeyStorePath).toURI().toURL(), "changeit", "jks");
+        
+        System.out.println("*********** SETUP DONE ***************************\n");
+    }
+
+    @After
+    public void tearDown() {
+        List<String> cmd = new ArrayList<>();
+
+        cmd.add("set");
+        cmd.add("configs.config.server-config.security-service.auth-realm.certificate.property.common-name-as-principal-name=false");
+        
+        CliCommands.payaraGlassFish(cmd);
+        
+        webClient.getCookieManager().clearCookies();
+        webClient.close();
+        
+        restartContainer("domain1");
+        
+        System.out.println("\n*********** TEST END ***************************\n");
+    }
+
+    @Test
+    public void testGetWithCorrectCredentials() throws Exception {
```

Comment:
Matt already lectured me on my unit tests, but they should follow this format: when_CONDITION_expect_RESULT (https://payara.atlassian.net/wiki/spaces/PORTAL/pages/754090391/Unit+testing)

Reply:
There's also a format for integration tests but haven't look closely at them

Reply:
It's a clone of an already existing test in the ee7 samples, don't know if that follows a different format as its a shared repo but I'll look into it

Reply:
Probably best to keep it consistent with the rest of ee7 samples